Ports/SuperTuxKart/patches/0002-Remove-reserved-word-from-goto-in-mcpp.patch

```diff
@@ -0,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Dominika Liberda <ja@sdomi.pl>
+Date: Sun, 2 Jul 2023 05:50:58 +0200
+Subject: [PATCH] Remove reserved word from goto in mcpp
+
+---
+ lib/mcpp/system.c | 10 +++++-----
+ 1 file changed, 5 insertions(+), 5 deletions(-)
+
+diff --git a/lib/mcpp/system.c b/lib/mcpp/system.c
+index a5b0d85c74203bd26365d99a874e5517aa78d2fb..1a61d240fec52c9cc96960566a03cca09084216c 100644
+--- a/lib/mcpp/system.c
++++ b/lib/mcpp/system.c
+@@ -3435,7 +3435,7 @@ search:
+     if (! fullname)                 /* Non-existent or directory    */
+         return  FALSE;
+     if (standard && included( fullname))        /* Once included    */
+-        goto  true;
++        goto  foo;
```

Comment:
Is there a compiler flag we can use for this file specifically (a la set_`source_files_properties(system.c PROPERTIES COMPILE_FLAGS -fallow-weird-goto-targets)` or w/e) to not make it complain about this file with our compiler version? This feels like a flags or an upstream issue tbh.

Reply:
.... ok looks like this mcpp library is abandonware, the latest release on sourceforge is from like 2008. So maybe this is fine :\

Reply:
Never mind, this is a C2X thing. Compiling the file with C17 should work

https://godbolt.org/z/3zoE1vvje

Reply:
ok, I got the fix.

add `-DCMAKE_C_STANDARD=17` to the configopts, and yeet this patch.

Reply:
Reported upstream, but we can work around it by telling CMake to use -std=c17 for all .c files with -DCMAKE_C_STANDARD 

<https://github.com/supertuxkart/stk-code/issues/4885>

Reply:
changing the standard to C17 doesn't actually fix the issue for me. It persist no matter if I add `-DCMAKE_C_STANDARD=17` or if I directly change lib/mcpp/CMakeFiles.txt (`set (CMAKE_C_FLAGS "-std=c17")` - it really doesn't wanna work.


I think the best bet for now would be to leave the patch as-is. this piece of code is easy to fix (and really bad) anyways..

Reply:
That's strange, it worked for me on Ubuntu. But you're right, this code was fragile when it was written. Thanks for taking a look at all my suggestions 😅